velox/functions/sparksql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -291,5 +291,14 @@ TEST_F(DateTimeFunctionsTest, dateSub) {
   EXPECT_EQ(parseDate("5881580-07-11"), dateSub("1969-12-31", kMin));
 }
 
+TEST_F(DateTimeFunctionsTest, dayOfWeek) {
+  const auto day = [&](std::optional<int32_t> date) {
+    return evaluateOnce<int64_t, int32_t>("dayofweek(c0)", {date}, {DATE()});
+  };
+  EXPECT_EQ(std::nullopt, day(std::nullopt));
+  EXPECT_EQ(5, day(parseDate("2009-07-30")));
+  EXPECT_EQ(4, day(parseDate("2023-08-23")));
```

Comment:
can we have some tests using timestamps/timezones as well?

Reply:
Fixed